api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -82,6 +76,45 @@ type ClientTrafficPolicySpec struct {
 	//
 	// +optional
 	HTTP1 *HTTP1Settings `json:"http1,omitempty"`
+	// HeaderSettings provides configuration for header management.
+	//
+	// +optional
+	Headers *HeaderSettings `json:"headers,omitempty"`
+}
+
+// ServerHeaderTransformation specifies the transformation required for the Server header
+// +kubebuilder:validation:Enum=Overwrite;AppendIfAbsent;PassThrough
+type ServerHeaderTransformation string
+
+const (
+	// ServerHeaderOverwrite causes the Server header to be overwritten with the configured
+	// ServerName
+	ServerHeaderOverwrite ServerHeaderTransformation = "Overwrite"
+	// ServerHeaderAppendIfAbsent causes the ServerName value to be used only if there is no
+	// Server header
+	ServerHeaderAppendIfAbsent ServerHeaderTransformation = "AppendIfAbsent"
+	// ServerHeaderPassThrough will not modify the value of the Server header.
+	ServerHeaderPassThrough ServerHeaderTransformation = "PassThrough"
+)
+
+// HeaderSettings providess configuration options for headers on the listener.
+type HeaderSettings struct {
```

Comment:
we discussed this in the community meeting today, here are some follow up questions
* can this use case we solved with the ResponseHeaderFilter ?
https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPHeaderFilter
* can the `suppressEnvoyHeaders` be represented with another name closer to the goal of the field such as `passThrough` etc ?

Reply:
> we discussed this in the community meeting today, here are some follow up questions
> 
> * can this use case we solved with the ResponseHeaderFilter ?
>   https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPHeaderFilter

No. The `ResponseHeaderFilter` unconditionally adds/overwrites a header whose value is given in the filter. This feature's `passThrough` option will keep the value of the Server header provided by the upstream, and there's no way to express the `appendIfAbsent` option with `ResponseHeaderFilter`.

> * can the `suppressEnvoyHeaders` be represented with another name closer to the goal of the field such as `passThrough` etc ?

You meant `serverHeaderTransformation`, not `suppressEnvoyHeaders`, right? `suppressEnvoyHeaders` has nothing to do with passing anything through and isn't the topic of this PR.

I used the name of the underlying feature in Envoy Proxy for the `serverHeaderTransformation` field. I think `passThrough` isn't a good name - it's only one of three options, and is exactly the opposite of the `overwrite` option.

 I'm open to any other name you have that you think fits better.

Reply:
from a user perspective, (correct me if im wrong), what you're really trying to solve here is - hey envoy, dont add any new response headers like `x-envoy-*` or `server`
we should come up with a name for this knob to define this opt in behavior

Reply:
how about a knob called `addEnvoyResponseHeaders`, making adding envoy headers along with `server` as opt in, so we supressby default

Reply:
maybe https://github.com/istio/api/pull/2240 will give a little help.

Reply:
thanks for sharing this @zirain, this is super useful
how about removing this API and disabling
* x-envoy-*
* server
by default

and we can introduce APIs in the future to opt into this

Reply:
fine to me

Reply:
So, to summarise the above:

1. The `HeaderSettings` structure is removed, and the default behaviour becomes that the `X-Envoy-` headers are removed and the Server header is defined as `passthrough`. This is done to try to effectively hide the presence of an Envoy Proxy between the client and the upstream target.
2. There is no support at this time for allowing the `X-Envoy` headers. This makes it harder for developers to perform certain tasks, for example when trying to measure latency with the [x-envoy-upstream-service-time](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/router_filter#x-envoy-upstream-service-time), or debugging various rewrite rules with the [x-envoy-original-path](https://www.envoyproxy.io/docs/envoy/v1.28.0/configuration/http/http_filters/router_filter#x-envoy-original-path) header.

How to proceed? Should I close this PR and create a new PR removing the `suppressEnvoyHeaders` option and implementing the behavioural changes listed here?

Reply:
hey @liorokman, agree with 1. 
for 2. we can make a opt in API e.g.
```
headers:
  enableEnvoyHeaders: true
``

Reply:
Updated to reflect this thread.

This PR changes the API and keeps the project building, a different PR will actually implement the feature.